pool/pool_manager.gno

```diff
@@ -39,82 +141,56 @@ func CreatePool(
 	token0Path string,
 	token1Path string,
 	fee uint32,
-	_sqrtPriceX96 string, // uint256
+	_sqrtPriceX96 string,
 ) {
 	common.IsHalted()
 	en.MintAndDistributeGns()
 
-	if token0Path == token1Path {
-		panic(addDetailToError(
-			errDuplicateTokenInPool,
-			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected token0Path(%s) != token1Path(%s)", token0Path, token1Path),
-		))
-	}
-
-	// wrap first
-	if token0Path == consts.GNOT {
-		token0Path = consts.WRAPPED_WUGNOT
-	} else if token1Path == consts.GNOT {
-		token1Path = consts.WRAPPED_WUGNOT
-	}
-
-	// then check if token0Path == token1Path
-	if token0Path == token1Path {
-		panic(addDetailToError(
-			errDuplicateTokenInPool,
-			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected token0Path(%s) != token1Path(%s)", token0Path, token1Path),
-		))
-	}
-
-	sqrtPriceX96 := u256.MustFromDecimal(_sqrtPriceX96)
+	poolInfo := newPoolInput(token0Path, token1Path, fee, _sqrtPriceX96)
 
-	if token1Path < token0Path {
-		panic(addDetailToError(
-			errTokenSortOrder,
-			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected token0Path(%s) < token1Path(%s)", token0Path, token1Path),
-		))
+	token0Path, token1Path, err := poolInfo.filterValidTokenPaths()
+	if err != nil {
+		panic(err)
 	}
 
-	// check tickSpacing for fee
-	tickSpacing := feeAmountTickSpacing[fee]
-	// calculate poolPath
 	poolPath := GetPoolPath(token0Path, token1Path, fee)
 
+	// TODO: make this as a parameter
+	prevAddr, prevRealm := getPrev()
+
 	// check whether the pool already exist
-	pool, exist := pools[poolPath]
+	pool, exist := pools.Get(poolPath)
 	if exist {
 		panic(addDetailToError(
 			errPoolAlreadyExists,
 			ufmt.Sprintf("pool_manager.gno__CreatePool() || expected poolPath(%s) not to exist", poolPath),
 		))
 	}
 
-	prevAddr, prevRealm := getPrev()
-	if !exist {
-		gns.TransferFrom(a2u(std.PrevRealm().Addr()), a2u(consts.PROTOCOL_FEE_ADDR), poolCreationFee)
-		std.Emit(
-			"PoolCreationFee",
-			"prevAddr", prevAddr,
-			"prevRealm", prevRealm,
-			"internal_poolPath", poolPath,
-			"internal_tokenPath", consts.GNS_PATH,
-			"internal_amount", ufmt.Sprintf("%d", poolCreationFee),
-		)
-
-		pool = newPool(token0Path, token1Path, fee, tickSpacing, sqrtPriceX96)
-		pools[poolPath] = pool
-
-		std.Emit(
-			"CreatePool",
-			"prevAddr", prevAddr,
-			"prevRealm", prevRealm,
-			"token0Path", token0Path,
-			"token1Path", token1Path,
-			"fee", ufmt.Sprintf("%d", fee),
-			"sqrtPriceX96", _sqrtPriceX96,
-			"internal_poolPath", poolPath,
-		)
-	}
+	gns.TransferFrom(a2u(std.PrevRealm().Addr()), a2u(consts.PROTOCOL_FEE_ADDR), poolCreationFee)
```

Comment:
`poolCreationFee` is var, not consts.
Do you think we need to check if amount > 0 execute transfer, if not don't

Reply:
I Agree. that can be prevent unnecessary transfers.

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/379/commits/8aade0fbf02500a7a86c5e86a1d33a483b8ac873